usecase/src/test/java/org/mobilitydata/gtfsvalidator/usecase/DownloadArchiveFromNetworkTest.java

```diff
@@ -0,0 +1,204 @@
+import org.mobilitydata.gtfsvalidator.usecase.port.RawFileRepository;
+import org.mobilitydata.gtfsvalidator.usecase.port.ValidationResultRepository;
+import org.mobilitydata.gtfsvalidator.usecase.utils.CustomFileUtils;
+import org.mockito.ArgumentCaptor;
+import org.mockito.ArgumentMatchers;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.*;
+
+class DownloadArchiveFromNetworkTest {
+
+    @Test
+    void downloadDataFromValidUrlShouldNotGenerateNotice() throws IOException {
+        ValidationResultRepository mockResultRepo = mock(ValidationResultRepository.class);
+        ExecParamRepository mockExecParamRepo = mock(ExecParamRepository.class);
+        when(mockExecParamRepo.hasExecParamValue(anyString())).thenReturn(true);
+        when(mockExecParamRepo.getExecParamValue(ArgumentMatchers.eq(ExecParamRepository.INPUT_KEY)))
+                .thenReturn(
+                        Path.of("./src/test/resources/download_archive_from_network_test_directory")
+                                .toString());
+        when(mockExecParamRepo.getExecParamValue(ArgumentMatchers.eq(ExecParamRepository.URL_KEY)))
+                .thenReturn("https://octa.net/current/google_transit.zip");
+
+        Logger mockLogger = mock(Logger.class);
+
+        CustomFileUtils mockFileUtils = mock(CustomFileUtils.class);
+        when(mockFileUtils.sizeOf(any(), anyString())).thenReturn(40L);
+
+        Path mockPath = mock(Path.class);
+
+        DownloadArchiveFromNetwork underTest =
+                new DownloadArchiveFromNetwork(mockResultRepo, mockExecParamRepo, mockLogger, mockFileUtils, mockPath);
+
+        underTest.execute();
+
+        verifyNoInteractions(mockResultRepo);
+        verify(mockExecParamRepo, times(1))
+                .getExecParamValue(ArgumentMatchers.eq(ExecParamRepository.URL_KEY));
+        verify(mockExecParamRepo, times(1))
+                .hasExecParamValue(ArgumentMatchers.eq(ExecParamRepository.URL_KEY));
+        verify(mockExecParamRepo, times(1))
+                .getExecParamValue(ArgumentMatchers.eq(ExecParamRepository.INPUT_KEY));
+        verify(mockFileUtils, times(1)).sizeOf(mockPath, CustomFileUtils.MEGABYTES);
+        verify(mockLogger, times(1)).info("Downloading archive");
+        verifyNoMoreInteractions(mockExecParamRepo, mockLogger, mockFileUtils, mockPath);
+
+        // remove created files
+        File toDelete = new File(
+                Path.of("./src/test/resources/download_archive_from_network_test_directory").toString());
+        assertTrue(toDelete.delete());
+    }
+
+    @Test
+    void invalidUrlShouldGenerateAddCannotDownloadArchiveFromNetworkNoticeToResultRepoAndThrowException() {
```

Comment:
ShouldGenerate or ShouldAdd, ShouldGenerateAdd at the moment.

Reply:
or ShouldGenerateAndAdd

Reply:
I opt for the second option you suggest, its better for clarity I think. Thanks :)